docs/features/records.md

```diff
@@ -190,7 +287,7 @@ public struct Pair : IEquatable<Pair>
     }
     public override bool Equals(object other)
     {
-		return (other as Pair?)?.Equals(this) == true;
+		return this.Equals(other as Pair);
```

Comment:
This code looks invalid to me -- `as` currently only works on `Nullable` or reference types. Has `as` changed or is this not meant to be valid code?